src/gui/SampleLoader.cpp

```diff
@@ -86,8 +87,21 @@ QString SampleLoader::openWaveformFile(const QString& previousFile)
 		previousFile.isEmpty() ? ConfigManager::inst()->factorySamplesDir() + "waveforms/10saw.flac" : previousFile);
 }
 
-std::unique_ptr<SampleBuffer> SampleLoader::createBufferFromFile(const QString& filePath)
+std::unique_ptr<SampleBuffer> SampleLoader::createBufferFromFile(const QString& filePath, bool collectErrorWhenNotFound)
 {
+	if (filePath.isEmpty()) { return std::unique_ptr<SampleBuffer>(); }
```

Comment:
```suggestion
	if (filePath.isEmpty()) { return nullptr; }
```
This is simpler and can't be confused with `std::make_unique<SampleBuffer>()`

Reply:
The original ``SampleBuffer`` bailed with having one frame on failure to load an audio file, so we can't return ``nullptr`` because some parts of the codebase expect some kind of buffer.

I could make the change, but I would have to ensure that the other parts of the codebase are handling buffers that are ``nullptr`` properly, which leaves a lot of room for mistakes to happen.

A good example would be ``SampleClip``. We would have to ensure the buffer is not ``nullptr`` inside ``Sample`` before returning from the function. And I shouldn't be doing the checks within ``Sample`` because not every part of the codebase uses ``Sample``, with some just using ``SampleBuffer``.